go/cmd/dolt/commands/sqlserver/sqlserver.go

```diff
@@ -216,6 +238,79 @@ func GetServerConfig(dEnv *env.DoltEnv, apr *argparser.ArgParseResults) (ServerC
 	return getCommandLineServerConfig(dEnv, apr)
 }
 
+// SetupDoltConfig updates the given server config with where to create .doltcfg directory
+func SetupDoltConfig(dEnv *env.DoltEnv, apr *argparser.ArgParseResults, config ServerConfig) error {
```

Comment:
Would be nice to extract this duplicate logic into the command package so that both commands can run the same logic